src/app/JarsList/JarsList.tsx

```diff
@@ -31,9 +31,15 @@ const JarItem = ({ jar, isSelected, onClick }: JarItemProps) => {
     logo,
   } = jar;
 
-  const copyJarLink = (ev: React.MouseEvent<HTMLSpanElement>) => {
+  const [copyClicked, setCopyClicked] = useState(false);
+  const handleClickCopy = (ev: React.MouseEvent<HTMLSpanElement>) => {
     ev.stopPropagation();
-    navigator.clipboard.writeText(url);
+    const timeout = setTimeout(() => {
+      navigator.clipboard.writeText(url);
+      setCopyClicked((current) => !current);
+    }, 300);
+    setCopyClicked((current) => !current);
+    return () => clearTimeout(timeout);
   };
```

Comment:
No need for this syncing. You can just copy link into clipboard and start navigation after that. It won't break anything and will make code cleaner.

And in general, when I need to wait for something, I usually use this util:

```ts
await new Promise(r => setTimeout(r, 300))
```

And you can move it to separate function if you want:

```ts
const waitFor = async (ms: number) => new Promise(r => setTimeout(r, ms))
```

Smaller. Cleaner